### PR TITLE
CLOUD-497 Add options parsing to psmdb

### PR DESCRIPTION
**Pull request by nonemax**

### CODE REVIEW

File: dbaas-lib/k8s/cmd.go [hunk 1]
```diff
@@ -94,19 +94,40 @@ func New(environment string) (*Cmd, error) {
...
+			cli.Env = append(cli.Env, "KUBECONFIG="+p.environment)
+		}
+		o, err = cli.CombinedOutput()
+		if err != nil {
+			if strings.Contains(string(o), "Unable to connect to the server") && i < n {
+				continue
+			}
+		}
+		break
+	}
```
Comment (dAdAbird) [inline dbaas-lib/k8s/cmd.go:126]: Shouldn't be here some sleep?

File: dbaas-lib/options/options.go [hunk 2]
```diff
@@ -20,27 +20,48 @@ func Parse(to interface{}, typ reflect.Type, options string) error {
...
+		startIndex := strings.Index(s, "[")
+		endIndex := strings.Index(s, "]")
+		if startIndex > 0 && endIndex > 0 {
+			newVal := s[:startIndex] + s[endIndex+1:]
+			return newVal
+		}
+	}
 
+	return s
+}
```
Comment (dAdAbird) [inline dbaas-lib/options/options.go:62]: 1. `if strings.Contains(s, "[") && strings.Contains(s, "]")` is redundant. Checking `startIndex` and `endIndex` is enough.
2. `if startIndex > 0` means the string starting from the bracket won't be processed. Is it intentional?
3. I don't really understand what this func is intended to do. _I see what it does, but for what?_ It worth to add more test cases

File: dbaas-lib/options/options.go [hunk 3]
```diff
@@ -136,7 +165,35 @@ func parseSliceValue(s string, refValue reflect.Value) (reflect.Value, error) {
 	return value, nil
 }
 
+func parseStructSliceValue(s string, refValue reflect.Value, fieldName string) (reflect.Value, error) {
+	value := refValue
+	//log.Println("type is:", refValue.Addr().Type().Elem().Elem())
```
Comment (dAdAbird) [inline dbaas-lib/options/options.go:170]: remove debug comments pls

File: dbaas-lib/options/options.go [hunk 4]
```diff
@@ -52,6 +73,12 @@ func setValue(val reflect.Value, value string) error {
 	default:
 		// TODO: maps, slices
```
Comment (dAdAbird) [inline dbaas-lib/options/options.go:74]: looks like this todo doesn't make sense anymore

File: dbaas-lib/options/options.go [hunk 5]
```diff
@@ -52,6 +73,12 @@ func setValue(val reflect.Value, value string) error {
...
 		return errors.Errorf("type %v not implemented", val.Kind())
+	case reflect.Struct:
+	/*v, err := parseStructValue(value, val)
+	if err != nil {
+		return errors.Errorf("parse value %s: %v", val, err)
+	}
+	val.Set(v)*/
```
Comment (dAdAbird) [inline dbaas-lib/options/options.go:81]: Are you sure? why?

**Review by dAdAbird — CHANGES_REQUESTED** [review]
Need more tests cases for the options parsing

**Comment by AlekSi** [comment]
Do you still need that branch?

**Pull request merged**